File: Java/Array/Remove Duplicates Sorted Array.java
//https://www.geeksforgeeks.org/problems/remove-duplicate-elements-from-sorted-array/1
class Solution {
    // Function to remove duplicates from the given sorted array
    public int removeDuplicates(int[] a) {
        int n = a.length;
        if (n == 0) return 0;

        int j = 0;
        for (int i = 0; i < n - 1; i++) {
            if (a[i] != a[i + 1]) {
                a[j++] = a[i];
            }
        }
        a[j++] = a[n - 1];
        
        return j; // new length of array without duplicates
    }
}

File: Python/Array/Remove Duplicates Sorted Array.py
//https://www.geeksforgeeks.org/problems/remove-duplicate-elements-from-sorted-array/1
class Solution:
    def removeDuplicates(self, arr):
        # Code Here
        n = len(arr)
        if n == 0:
            return 0

        j = 0
        for i in range(1, n):
            if arr[i] != arr[i-1]:
                j += 1
                arr[j] = arr[i]

        return j + 1
